hibernate-reactive-core/src/test/java/org/hibernate/reactive/IdentityGeneratorTypeForCockroachDBTest.java

```diff
@@ -78,21 +76,23 @@ public void longIdentityType(TestContext context) {
 
 	@Test
 	public void integerIdentityType(TestContext context) {
-		thrown.expectMessage( "too big" );
-		thrown.expectMessage( "Integer" );
-
 		test( context, getMutinySessionFactory()
-				.withTransaction( (s, tx) -> s.persist( new IntegerTypeEntity() ) )
+				.withTransaction( (s, tx) -> s.persist( new IntegerTypeEntity() )  )
+				.onFailure().recoverWithUni( e -> {
+					ReactiveAssertions.assertMessage(e, "too big", "Integer" );
+					return Uni.createFrom().nullItem();
+				})
 		);
 	}
 
 	@Test
 	public void shortIdentityType(TestContext context) {
-		thrown.expectMessage( "too big" );
-		thrown.expectMessage( "Short" );
-
 		test( context, getMutinySessionFactory()
-				.withTransaction( (s, tx) -> s.persist( new ShortTypeEntity() ) )
+				.withTransaction( (s, tx) -> s.persist( new ShortTypeEntity() )  )
+				.onFailure().recoverWithUni( e -> {
```

Comment:
If you only check on failure, the test won't fail if the exception is not thrown.
You can use `assertThrown` and the check the message

Reply:
refactored and fixed